### PR TITLE
Network peering profile fix

### PR DESCRIPTION
**Pull request by Adelmar92**
## Description

Added profile to Network Peering

ContainerID now is mandatory

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] For CFN Resources: I have released by changes in the private registry and proved by change works in Atlas

### CODE REVIEW

File: cfn-resources/network-peering/cmd/resource/resource.go [hunk 1]
```diff
@@ -71,48 +71,31 @@ func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler
 		return *errEvent, nil
 	}
 
-	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
-	if err != nil {
-		_, _ = logger.Warnf("CreateMongoDBClient error: %s", err)
-		return progressevents.GetFailedEventByCode(fmt.Sprintf("Failed to Create Client : %s", err.Error()),
-			cloudformation.HandlerErrorCodeInvalidRequest), nil
+	if currentModel.Profile == nil {
```
Comment (andreaangiolillo) [inline cfn-resources/network-peering/cmd/resource/resource.go:74]:
```suggestion
	if currentModel.Profile == nil || currentModel.Profile == "" {
```

File: cfn-resources/network-peering/cmd/resource/resource.go [hunk 2]
```diff
@@ -26,6 +25,7 @@ import (
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
 	progressevents "github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
+	"github.com/openlyinc/pointy"
```
Comment (andreaangiolillo) [inline cfn-resources/network-peering/cmd/resource/resource.go:28]: Should we use pointy or the aws one?

**Review by andreaangiolillo — COMMENTED** [review]
LGTM once the tests are all green

File: cfn-resources/network-peering/cmd/resource/resource.go (hunk 2 above)
Comment (colm-quinn) [reply cfn-resources/network-peering/cmd/resource/resource.go:28]: Let's use the aws version for consistency.

File: cfn-resources/network-peering/cmd/resource/resource.go [hunk 3]
```diff
@@ -139,29 +122,24 @@ func Read(req handler.Request, prevModel, currentModel *Model) (handler.Progress
 		return *errEvent, nil
 	}
 
-	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
-	if err != nil {
-		_, _ = logger.Warnf("CreateMongoDBClient error: %s", err)
-		return progressevents.GetFailedEventByCode(fmt.Sprintf("Failed to Create Client : %s", err.Error()),
-			cloudformation.HandlerErrorCodeInvalidRequest), nil
+	if currentModel.Profile == nil {
```
Comment (colm-quinn) [inline cfn-resources/network-peering/cmd/resource/resource.go:125]:
```suggestion
	if currentModel.Profile == nil || currentModel.Profile == "" {
```

File: cfn-resources/network-peering/cmd/resource/resource.go [hunk 4]
```diff
@@ -188,15 +166,14 @@ func Update(req handler.Request, prevModel *Model, currentModel *Model) (handler
 	setup()
 	_, _ = logger.Debugf("Update currentModel:%+v", currentModel)
 
-	if errEvent := validateModel(UpdateRequiredFields, currentModel); errEvent != nil {
-		return *errEvent, nil
+	if currentModel.Profile == nil {
```
Comment (colm-quinn) [inline cfn-resources/network-peering/cmd/resource/resource.go:169]:
```suggestion
	if currentModel.Profile == nil || currentModel.Profile == "" {
```

File: cfn-resources/network-peering/cmd/resource/resource.go [hunk 5]
```diff
@@ -249,11 +226,15 @@ func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler
 	if errEvent := validateModel(DeleteRequiredFields, currentModel); errEvent != nil {
 		return *errEvent, nil
 	}
-	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
-	if err != nil {
-		_, _ = logger.Warnf("CreateMongoDBClient error: %s", err)
-		return progressevents.GetFailedEventByCode(fmt.Sprintf("Failed to Create Client : %s", err.Error()),
-			cloudformation.HandlerErrorCodeInvalidRequest), nil
+
+	if currentModel.Profile == nil {
```
Comment (colm-quinn) [inline cfn-resources/network-peering/cmd/resource/resource.go:230]: Stopping here, can we update all the usages using the example above with both checks.

File: cfn-resources/network-peering/docs/README.md [hunk 6]
```diff
@@ -101,30 +99,18 @@ _Type_: String
 
 _Update requires_: [Replacement](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-update-behaviors.html#update-replacement)
 
-#### ConnectionId
+#### Profile
 
-Unique 24-hexadecimal digit string that identifies the MongoDB Cloud network container that contains the specified network peering connection.
+Profile used to provide credentials information, (a secret with the cfn/atlas/profile/{Profile}, is required), if not provided default is used
```
Comment (colm-quinn) [inline cfn-resources/network-peering/docs/README.md:104]:
```suggestion
The profile is defined in AWS Secret manager. See [Secret Manager Profile setup](../../../examples/profile-secret.yaml).
```

File: cfn-resources/network-peering/mongodb-atlas-networkpeering.json [hunk 7]
```diff
@@ -56,46 +44,51 @@
             "description": "Unique 24-hexadecimal digit string that identifies the network peering connection that you want to retrieve.",
             "type": "string"
         },
-        "ApiKeys": {
-            "$ref": "#/definitions/apiKeyDefinition"
+        "Profile": {
+            "type": "string",
+            "description": "Profile used to provide credentials information, (a secret with the cfn/atlas/profile/{Profile}, is required), if not provided default is used",
```
Comment (colm-quinn) [inline cfn-resources/network-peering/mongodb-atlas-networkpeering.json:49]:
```suggestion
            "description": "The profile is defined in AWS Secret manager. See [Secret Manager Profile setup](../../../examples/profile-secret.yaml).",
```

**Review by colm-quinn — DISMISSED** [review]
LGTM, thanks for all the updates

**Review by colm-quinn — APPROVED** [review]

**Pull request merged**